data-prepper-logstash-configuration/src/main/java/com/amazon/dataprepper/logstash/model/LogstashAttribute.java

```diff
@@ -6,22 +6,38 @@
  * @since 1.2
  */
 public class LogstashAttribute {
-    private String attributeName;
-    private LogstashAttributeValue attributeValue;
+    private final String attributeName;
+    private final LogstashAttributeValue attributeValue;
 
     public String getAttributeName() {
         return attributeName;
     }
 
-    public void setAttributeName(String attributeName) {
-        this.attributeName = attributeName;
-    }
-
     public LogstashAttributeValue getAttributeValue() {
         return attributeValue;
     }
 
-    public void setAttributeValue(LogstashAttributeValue attributeValue) {
-        this.attributeValue = attributeValue;
+    private LogstashAttribute (LogstashAttributeBuilder builder) {
+        this.attributeName = builder.attributeName;
+        this.attributeValue = builder.attributeValue;
```

Comment:
Do we have any requirements on these values? Can they be null or do we need to enforce non-null?

The same question applies to the other models.